src/shared/projectConfiguration.ts

```diff
@@ -0,0 +1,67 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as crypto from 'crypto';
+import { machineIdSync } from 'node-machine-id';
+import { PlatformInformation } from './platform';
+import { ITelemetryReporter, getTelemetryProps } from './telemetryReporter';
+import { DotnetInfo } from './utils/dotnetInfo';
+
+export interface ProjectConfigurationMessage {
+    ProjectId: string;
+    SessionId: string;
+    OutputKind: number;
+    ProjectCapabilities: string[];
+    TargetFrameworks: string[];
+    References: string[];
+    FileExtensions: string[];
+    FileCounts: number[];
+    SdkStyleProject: boolean;
+}
+
+export function reportProjectConfigurationEvent(
+    reporter: ITelemetryReporter,
+    projectConfig: ProjectConfigurationMessage,
+    platformInfo: PlatformInformation,
+    dotnetInfo: DotnetInfo | undefined,
+    solutionPath?: string,
+    useModernNet?: boolean
+) {
+    let solutionId = '';
+    if (solutionPath) {
+        solutionId = createSolutionId(solutionPath);
+    }
+
+    const telemetryProps = getTelemetryProps(platformInfo);
+    telemetryProps['SolutionId'] = solutionId;
+    telemetryProps['ProjectId'] = projectConfig.ProjectId;
+    telemetryProps['SessionId'] = projectConfig.SessionId;
+    telemetryProps['OutputType'] = projectConfig.OutputKind?.toString() ?? '';
+    telemetryProps['ProjectCapabilities'] = projectConfig.ProjectCapabilities?.join(' ') ?? '';
+    telemetryProps['TargetFrameworks'] = projectConfig.TargetFrameworks.join('|');
+    telemetryProps['References'] = projectConfig.References.join('|');
+    telemetryProps['FileExtensions'] = projectConfig.FileExtensions.join('|');
+    telemetryProps['FileCounts'] = projectConfig.FileCounts?.join('|') ?? '';
+    telemetryProps['NetSdkVersion'] = dotnetInfo?.Version ?? '';
+    telemetryProps['sdkStyleProject'] = projectConfig.SdkStyleProject.toString();
+
+    if (useModernNet) {
+        telemetryProps['useModernNet'] = useModernNet.toString();
+    }
```

Comment:
Just to note this flag is being obsoleted with the OOP work. It's still useful for Omnisharp telemetry maybe but isn't useful for the LSP case. Just something to call out if anybody is actually using this.

Reply:
It doesn't appear to be used in the dotnet data, so think we're good to remove if you need to for OOP